parlai/tasks/convai2/worlds.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from copy import deepcopy
+
+from parlai.core.params import ParlaiParser
+from parlai.core.agents import create_agent
+from parlai.core.worlds import create_task
+from parlai.core.worlds import DialogPartnerWorld, validate
+from parlai.agents.repeat_label.repeat_label import RepeatLabelAgent
+
+
+class InteractiveWorld(DialogPartnerWorld):
+    def __init__(self, opt, agents, shared=None):
+        super().__init__(opt, agents, shared)
+        print("[ loading personas.. ]")
+        self.load_personas()
+
+    def load_personas(self):
+        # Create ConvAI2 data so we can assign personas.
+        convai2_opt = self.opt.copy()
+        convai2_opt['task'] = 'convai2:both'
+        convai2_opt['interactive_task'] = False
+        convai2_agent = RepeatLabelAgent(convai2_opt)
+        self.convai2_world = create_task(convai2_opt, convai2_agent)
+        self.cnt = 0
+
+    def get_new_personas(self):
+        # Find a new episode
+        while True:
+            self.convai2_world.parley()
+            msg = self.convai2_world.get_acts()[0]
+            if msg.get('episode_done', False):
+                self.convai2_world.parley()
+                msg = self.convai2_world.get_acts()[0]
+                break
+        txt = msg.get('text', '').split('\n')
+        a1_persona = ""
+        a2_persona = ""
+        for t in txt:
+            if t.startswith("partner's persona:"):
+                a1_persona += t.replace("partner's persona:", 'your persona:') + '\n'
+            if t.startswith('your persona:'):
+                a2_persona += t + '\n'
+        print("Enter [DONE] if you want a new partner at any time.\n")
```

Comment:
it's not clear to me where the search for `[DONE]` actually lives

Reply:
in local_human if it is being used.. so I moved this print statement there